### ISSUE TITLE
Add KeybindingService.resolve(keybinding: Keybinding, target: IContextKeyServiceTarget)

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#14603]
Extracted from #8818

**Comment by Tyriar** [comment]
@alexandrudima how would this be used? I was expecting something like `IKeybindingService.lookupKeybindings` that also returned the command's `ContextKeyExpr`?

The `Keybinding` object itself is ambiguous and could represent many commands, I only care about one particular one.

**Comment by alexandrudima** [comment]
@Tyriar Sorry that I am again confused. I thought your code would look like this:

```
this._xterm.attachCustomKeydownHandler((event: KeyboardEvent) => {
    var kbEvent = new StandardKeyboardEvent(event);
    var commandId = keybindingService.resolve(kbEvent.asKeybinding(), myXtermDomNode)
    // commandId now holds the command that would execute if this even would reach us
    if (oneOfMyCommands.indexOf(commandId)) {
        return false;
    }
});
```

So the keybinding service will evaluate the keybinding in the context of your dom node and tell you which command will execute should that keybinding actually reach it.

**Comment by Tyriar** [comment]
That should work :+1:

**Comment by Tyriar** [comment]
I merged the change we worked on together in #8818
